### ISSUE TITLE
keyObject is undefined when executing keyObject.extractable

### ISSUE THREAD

**Issue by marcato15** [microsoft/MSR-JavaScript-Crypto#13]
There is an issue with how the code handles `raw` formatted keys.  `keyObject` is not defined  on lines 7831 and  8157. If I add the following block that is used elsewhere the code works fine but I am not a crypto expert so not sure if there is something else that should be done.

````
var   keyObject = {
                        k: msrcryptoUtilities.toArray(p.keyData)
                    };
````

**Comment by marcato15** [comment]
I'm not 100% sure. I am using https://github.com/aws/aws-encryption-sdk-javascript library and configuring this library as a fallback for the sublte crypto functions the SDK needs since I am running this in a React Native project. I assume it was during the import key functionality of the  SDK as it was encrypting/decrypting.

**Comment by ljoy913** [comment]
I merged a fix with an update version 1.6.5
Thanks to @[misbaharefin](https://github.com/misbaharefin) for submitting a PR to fix this same issue
I'll leave this open until you get a chance to verify

**Comment by marcato15** [comment]
Thanks! That seems to do it. 

After I downloaded the latest version I was reminded of another issue I had fixed earlier.  I am running this in React Native and so the "browser" environment there is slightly different than a normal one. As a result the section of code on lines 75-77 causes an issue since `self` is defined but not `self.location`. I just added an additional check to that else if in my code to get around the issue. I can open a separate issue if need be since its unrelated to the above fix, which is working great now. 

````
    } else if (typeof self !== "undefined" && typeof self.location !== "undefined") {
                return self.location.href;
            }
````
